app/models/storage.rb

```diff
@@ -70,6 +71,12 @@ class Storage < ApplicationRecord
 
   SUPPORTED_STORAGE_TYPES = %w( VMFS NFS FCP ISCSI GLUSTERFS )
 
+  supports :smartstate_analysis do
+    if ext_management_systems.blank? || !ext_management_system.kind_of?(ManageIQ::Providers::Vmware::InfraManager)
```

Comment:
As far as I know, we should be using the Datastore browser for VMware. ESXi hosts don't have a console OS, so `ls -l` would not be possible.